File: EESoft/DataCenter/DataCenter - Server/Forms/WizradFrm.cs
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Collections;
using Microsoft.VisualBasic;
using System.Collections.Generic;
using System;
using System.IO;

namespace DataCenter_Server
{
	public partial class WizradFrm
	{
		public WizradFrm()
		{
			InitializeComponent();
		}
		private short Int1;
		public void WizradFrm_FormClosing(object sender, FormClosingEventArgs e)
		{
			MainModule.FrmMainForm = new MainForm();
			MainModule.FrmMainForm.Show();
        }
		public void BtnCancel_Click(Object sender, EventArgs e)
		{
			this.Close();
		}
		public void TxtName_TextChanged(Object sender, EventArgs e)
		{
			if (TxtName.Text.Trim().Length > 0)
				BtnAdd.Enabled = true;
			else
				BtnAdd.Enabled = false;
            TextBox TB = (TextBox)sender;
            for (Int1 = 0; Int1 <= TB.Text.Trim().Length - 1; Int1++)
			{
                if (!char.IsLetterOrDigit(TB.Text.Trim(), Int1) && TB.Text[Int1].ToString() != "_" && TB.Text[Int1].ToString() != " ")
                {
                    TB.Text = "";
                    //Prepare the Error MSG
                    MainModule.MsgBox_("ãä ÝÖáß ÇÏÎá ÍÑæÝ æ ÇÑÞÇã", 98, 23, false, this);
                    break;
                }
			}
		}
		public void BtnAdd_Click(Object sender, EventArgs e)
		{
			if (Interaction.MsgBox("هل انت متأكد؟", MsgBoxStyle.YesNo, "انشـــــــــاء") == MsgBoxResult.No)
				return;
            string NewFolderErrorResult = MainModule.FunLib.InsertNewFolder(TxtName.Text.Trim(), TxtName.Text.Trim(), DateTime.Now.Date.ToString(), MainModule.AdminID, "", "0");
            if (NewFolderErrorResult != null)
            {
                Directory.CreateDirectory(String.Format("{0}\\{1}", MainModule.ProjectsPath, TxtName.Text.Trim()));
                Close();
            }
		}
		public void TxtLocation_TextChanged(Object sender, EventArgs e)
		{
			if (TxtName.Text.Trim().Length > 0)
				BtnAdd.Enabled = true;
			else
				BtnAdd.Enabled = false;
		}
	}
}

File: EESoft/DataCenterVB/Backup/DataCenter - Server/Forms/WizradFrm.vb
Public Class WizradFrm
    Private ErrorProvider As String = Nothing
    Private Int1 As Int16

    Private Sub WizradFrm_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
        FrmMainForm = New MainForm
        FrmMainForm.Show()
    End Sub
    Private Sub WizradFrm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        MainTreeView.ExpandAll()
        MainTreeView.Focus()
        MainTreeView.SelectedNode = MainTreeView.Nodes.Item(0).Nodes(4)
    End Sub
    Private Sub BtnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnCancel.Click
        Me.Close()
    End Sub
    Private Sub TxtName_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TxtName.TextChanged
        'If TxtName.Text.Trim.Length > 0 Then
        '    For Int1 = 0 To TxtName.Text.Trim.Length - 1
        '        If Char.IsLetterOrDigit(TxtName.Text.Trim.Chars(Int1)) = False Then
        '            TxtName.Clear()
        '        End If
        '    Next
        'End If
        If TxtName.Text.Trim.Length > 0 And TxtLocation.Text.Trim.Length > 0 Then
            BtnAdd.Enabled = True
        Else
            BtnAdd.Enabled = False
        End If
        For Int1 = 0 To sender.Text.Trim.Length - 1
            If Char.IsLetterOrDigit(sender.Text.Chars(Int1)) = False And sender.Text.Chars(Int1) <> "_" And sender.Text.Chars(Int1) <> " " Then
                sender.text = ""
                'Prepare the Error MSG
                MsgBox_("Please enter letter or digit", 98, 23, False, Me)
                Exit For
            End If
        Next
    End Sub
    Private Sub BtnAdd_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnAdd.Click
        If MsgBox("Are you sure ?", MsgBoxStyle.YesNo, "WARNING") = MsgBoxResult.No Then
            Exit Sub
        End If
        ErrorProvider = Nothing
        MainProgressBar.Visible = True
        Me.Enabled = False
        ErrorProvider = FunLib.AddNewProject(TxtName.Text.Trim, TxtLocation.Text.Trim, CDate(MCStart.SelectionEnd.ToString).ToShortDateString, CDate(MCEnd.SelectionEnd.ToString).ToShortDateString, AdminID, ProjectsPath, MainProgressBar.Value)
        If ErrorProvider = Nothing Then
            MsgBox_("All complete - New project added", 74, 23, True, Me)
            Me.Close()
        Else
            MsgBox_(ErrorProvider, 0, 23, False, Me)
        End If
        Me.Enabled = True
    End Sub
    Private Sub TxtLocation_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TxtLocation.TextChanged
        If TxtName.Text.Trim.Length > 0 And TxtLocation.Text.Trim.Length > 0 Then
            BtnAdd.Enabled = True
        Else
            BtnAdd.Enabled = False
        End If
    End Sub

End Class